test/Test/BaseDAO/Proposal.hs

```diff
@@ -129,7 +129,9 @@ nonUniqueProposal = uncapsNettest $ do
 
 voteValidProposal :: (Monad m) => NettestImpl m -> m ()
 voteValidProposal = uncapsNettest $ do
-  ((owner1, _), (owner2, _), dao, _) <- originateBaseDaoWithConfig @Integer @Empty () config
+  ((owner1, _), (owner2, _), dao, admin) <- originateBaseDaoWithConfig @Integer @Empty () config
+  callFrom (AddressResolved admin) dao (Call @"Set_voting_period") 120
+  callFrom (AddressResolved admin) dao (Call @"Set_quorum_threshold") 4
```

Comment:
I have tried to look into the source of the problem, please confirm with me if this sounds correct to you as well.

I noticed that by default in `BaseDAO.CLI` we set `votingPeriod` and `quorumThreshold` to the values you are setting here, however we don't do so for tests because in `originateBaseDaoWithBalance` the `#votingPeriod` and `#quorumThreshold` are different.

Looking at recent changes it seems this was introduced in [this commit](https://github.com/tqtezos/baseDAO/commit/530ec00df307c5de1e3a0339f5686c76be8aa253).

My question is: instead of adding these two calls in some of these tests, couldn't we modify the `originateData` in `originateBaseDaoWithBalance` to use these values instead?
Or is there another reason to not do this that I am missing?

Reply:
Some of the tests prefer short voting period since it makes it run faster. 
So I make it as the default and only set longer voting period if necessary.

The failing tests are not detected previously because I thought all time sensitive tests only run in emulator until batch operation is implemented in cleveland. 

But this is not the case, all the failing tests here are vote related tests, and they require longer voting period. They however, do not causes long running time (due to not having to use `advanceTime`), so I keep them in the real network tests and simply set a longer voting period.

Reply:
I see, saving time with shorter voting periods when possible makes sense.

However I'd still prefer to set this beforehand instead of making a call to set the values, I think it should be handled with configuration instead.

So I suggest this:
- in `originateBaseDaoWithBalance`'s `mkStorage` call instead of setting the `votingPeriod` to a default value of `20`, let's set it to the `cMinVotingPeriod` value from `config`
- then let's change the `Test.BaseDAO.ProposalConfig` `config` to set the `votingPeriod` to `20` instead of the 7 days from `DAO.defaultConfig`. (Note: I think we should do the same for the other configs in this module as well)
- create a new config in `Test.BaseDAO.ProposalConfig` for these voting tests, which is identical to `config` except having the `votingPeriod` set to `120`.

Plus, we do the same for the `quorumThreshold`.

This way we'd still have most tests to use the lower voting period and these ones to use the bigger one, but without making more calls for it. What do you think?
If you agree please make this in a separate commit, so we can check that there are indeed no problems in doing it this way and revert otherwise.

P.s. while we are at it, we could also change `dropProposal` and perhaps `flushProposalIntegrational` to not set these values with calls but by changing the config they are created with (this time in place)

Reply:
> - then let's change the `Test.BaseDAO.ProposalConfig` `config` to set the `votingPeriod` to `20` instead of the 7 days from `DAO.defaultConfig`. (Note: I think we should do the same for the other configs in this module as well)

I think you mean `minVotingPeriod` right, since votoingPeriod is stored in `Storage` instead of `Config`. Tho, if we use `minVotingPeriod`, it can still work since we use that field to set the storage. 

I think it would be nice to be able to set the field of the storage directly (instead of using `minVotingPeriod` field as a passer). This will require quite a lot of refactoring. 

So what's your advice on this, should we simply set `minVotingPeriod` in the config, or modify `originateBaseDao` to be able to accepts a custom storage.

Reply:
> I think you mean minVotingPeriod

Yeah, that's correct.

IMO we should do this: here let's use `minVotingPeriod` in `mkStorage` since it's easier to do and I think makes sense (these are small values and after all indeed these tests **need** a minimum value of `120`) and let's create a separate issue to consider and eventually make the changes to pass a different `votingPeriod` using the config.

In fact if you agree, please create the issue yourself and add a TODO mentioning it in `originateBaseDaoWithBalance` as well.